### DISCUSSION TITLE
Hilfe benötigt: Instabiltität im System: Verbindung DTU zu den Wechselrichtern

### DISCUSSION THREAD

**Discussion by 2000ede** [tbnobody/OpenDTU#1788] [Q&A] [answered]
Hallo!

Ich habe große Stabilitätsprobleme (an sich von Beginn an) und die Ursachenanalyse gestaltet sich als extrem schwierig.

- 2 bzw 3 Wechselrichter (HMS 1600) an der DTU
- DTU bekommt von außen alle x Sekunden (aktuell 6 Sekunden) neue Limite für die Wechselrichter (Nulleinspeisung).

Problem: Die in der DTU angezeigten Sekunden für DTU zu Wechselrichter sind nochmal aufgrund der bis zu 3 Wechselrichter 12 Sekunden. Im Problemfall laufen sie aber hoch bis über 100 Sekunden. Auch gehen die Limite dann hart auf 0. Oder die DTU friert ganz ein.

Wir haben allDann läuft es 2, 3 Tage ohne Probleme es durchprobiert, unzählige Konstellationen (DTU ausgetauscht, Limitsteuerung aus), Antennen perfekt ausgerichtet usw.
**Dann läuft es 2, 3 Tage ohne Probleme** und wie gestern geht nichts mehr.

Heute Nacht haben wir Wechselrichter vom Strom genommen. Also DC und AC war aus.
Heute morgen läuft wieder alles super.

Kann es an den Wechselrichtern liegen? Das da ein (Programm-)Speicher oder so vollläuft der sich nicht leert, wenn nachts AC an bleibt? Kann man da was anaylsieren oder einstellen?

Ob es daran liegt, wissen wir nicht. Aber wir haben wirklich schon alles ausprobiert.

Über Hilfe wären wir extrem dankbar!

**Comment by Marc9485** [comment]
Netzteil mit mehr Leistung an den esp und evtl. noch nen Kondensator ans nrf modul

**Reply by 2000ede** [reply]
Danke für den Hinweis. Bereits mehrfach getauscht. Auch DTU getauscht. 
Alles versucht nach dem Ausschlussprinzip. Mittlerweile denke ich es liegt an den Wechselrichtern.
Heute alles sehr gut, Sekunden immer im grünen Bereich. Allerdings 2 mal Wechselrichter nicht mehr produziert, obwohl er Limit sauber empfangen hat weiter. Erst Neustart wieder hat er produziert.

**Comment by MetaChuh** [comment]
@2000ede 

die opendtu hm und hms wechselrichter kommunikation via nrf24+ und cmt2300a sollte nahezu felsenfest funktionieren.
(siehe die opendtu issues und discussions bezüglich funk zuverlässigkeit erfahrung der user)

bisher waren die meisten probleme auf grund von wlan problemen.
spezifisch: solange kein nachbar zuhause war, waren alle tests erfolgreich, jedoch sobald alle nach hause kommen, ist der verwendete wlan kanal nicht mehr responsive, was bei den tests bis zu einem unerwarteten watchdog reboot der opendtu führt.
(siehe uptime der dtu bei der bei euch dies der fall ist)

hast du aufzeichnungen, zu welchen uhrzeiten es immer funktioniert hat, und zu welcher uhrzeit zeit es mehrfach zu ausfällen kam ?
dann können wir uns hoffentlich an deine spezifische lösungsaufgabe mit erfahrungswerten anpassen.

**Reply by 2000ede** [reply]
Hallo MetaChuh,

erstmal vielen Dank, dass du mir antwortest!
Ich bin echt am Verzweifeln, ich komme aus der Softwarebranche und Ausschlussprinzip und Logik sind eigentlich meine Steckenpferde bei der Fehlersuche. Ich hab hier echt schon rauf und runter alles probiert und ausgetauscht, Konstellationen gegenübergestellt usw.
Der Zeiteinsatz ist gigantisch und sprengt die Relation zum Eigentlichen. Aber auf der anderen Seite ist mein sportlicher Ehrgeiz geweckt.

Es geht hier um 2 bzw 3 HMS-1600 bzw 1800
OpenDTU als Komplettsatz gekauft, eine weitere zum Test/Vergleich
Ich habe auch ein kleines Logging mitlaufen bzw über Tasmota lasse ich mir es genau anzeigen.

Uhrzeiten usw. hab ich nicht (ich mache es für einen Bekannten, bin selber nicht vor Ort. Aber Zugriff.
Aber in der Tag Sonntag war (zufällig?) der Horrortag. Da gingen die Sekunden nur so hoch und einfrieren

Heute war alles sauber bis auf das zweimal ein unterschiedlicher Wechselrichter zwar Limite angenommen hat, aber der Ertrag auf 0 eingeforen ist. Erst Wechselrichterneustart brachte ihn wieder zum leben.

Ich habe in meiner Limit-Steuerung mittlerweile soviele Scheifen eingebaut, die auf Störungen reagiert und sie versucht zu beheben.
(Auch mit einer gekauften Steuerung traten die Probleme auf - sogar massiv mehr)

Wenn wirklich das WLan als Problem gesehen wird, würde ich prüfen, ob man der DTU nicht einen LAN Zugang gibt (weiss nicht ob das geht). Alles was das Problem weiter eingrenzt - ich bin dabei.

 Ich freue mich, wenn du am Ball bleibst und mir versuchst zu helfen! Dafür bin ich dir jetzt schon dankbar!
 Auch für Zugriffe usw. stehe ich zur Verfügung!

**Comment by MetaChuh** [comment]
hi @2000ede

> erstmal vielen Dank, dass du mir antwortest!

gerne, machen wir alle, sobald der eine oder der ander online ist.

> Der Zeiteinsatz ist gigantisch und sprengt die Relation zum Eigentlichen.

mache es nur für dich aus freude, sonst macht es wenig sinn.

> Uhrzeiten usw. hab ich nicht (ich mache es für einen Bekannten, bin selber nicht vor Ort)

no problem, geht nur um private erfahrungswerte austauschen, nicht dass du eine pflicht hast, oder verabsäumt hast etwas aufzuzeichnen ... vor allem nicht, wenn es lief, bevor du gegangen bist.

> Heute war alles sauber bis auf das zweimal ein unterschiedlicher Wechselrichter zwar Limite angenommen hat, aber der Ertrag auf 0 eingeforen ist. Erst Wechselrichterneustart brachte ihn wieder zum leben.

heute: montag, weniger menschen daheim.

eingefroren: das kennen wir, wenn das 2.4ghz wlan netz entweder dicht ist, oder in unmittelbarer nähe der dtu (lokal oder nachbar) gerade ein mikrowellenherd läuft (insider: sendung mit der maus spruch: klingt zwar blöd, is' aber so).

> Ich habe in meiner Limit-Steuerung mittlerweile soviele Scheifen eingebaut, die auf Störungen reagiert und sie versucht zu beheben.

sehr cool, in welcher programmiersprache ?

> Wenn wirklich das WLan als Problem gesehen wird, würde ich prüfen, ob man der DTU nicht einen LAN Zugang gibt (weiss nicht ob das geht). Alles was das Problem weiter eingrenzt - ich bin dabei.

ja, und ist bereits vor langer zeit geschehen.
opendtu unterstützt viele gängige lan module für esp, zb. olimex 32 poe, wt32 eth01, etc. um das wlan problem zu eliminieren.

> Ich freue mich, wenn du am Ball bleibst und mir versuchst zu helfen! Dafür bin ich dir jetzt schon dankbar!
Auch für Zugriffe usw. stehe ich zur Verfügung!

gerne, aber das bin nicht nur ich, sondern wir alle aus der dtu und solar community ... ich war nur gerade zufällig privat online.

**Reply by 2000ede** [reply]
Ich werde mich nun auf das Wlan als Hauptgrund fokkussieren. (bisher habe ich das ausgeschlossen, weil fettes WLAN vor Ort da ist und der Verkäufer der DTU sagte nach dem Sekundenzählen: Dicht an die Wechselrichter ran, hat zu wenig Kontakt.

Macht es Sinn, über /api/network/status rssi zu loggen oder per ping loggen Werte zu sammeln? Oder sagen die Werte nichts aus?

Meine Steuerung habe ich mit Hilfe von Gerhard in Tasmota Script gemacht. Ich bin zwar Anfänger, aber hartnäckig und das Zielbild immer vor Augen. So kostet mich fehlendes Anfängerwissen oft Stunden.  Ich verfolge auch die Strategie, alles mit dem Tasmota zu machen (auch Visualisierung) und nicht HA und co einzusetzen um u.a. auch Senioren einen einfachen Zugang zu geben.

P.S. leider komme ich nicht auf die neue DTU Version, siehe: https://github.com/tbnobody/OpenDTU/discussions/1760
(hatte verwechselt, dass hier zum Glück deutsch geschrieben werden darf).

Ich bleibe am Ball mit den Performanceproblemen und berichte hier gerne. Vielleicht helfe ich damit am Ende auch anderen.

Gerne stelle ich dir auch persönlich meine Löung vor.

**Reply by Marc9485** [reply]
Du hast doch probleme mit der Verbindung zu den Wechselrichtern!?! Dann kannst du WLAN als Fehler ausschließen!

**Comment by 2000ede** [comment, marked as answer]
Hallo @MetaChuh

dank deinem Hinweis habe ich das Problem zu 98% beheben können. 
Es lag an einem aus meiner Sicht nicht gut gebauten (fertig gebaut gekauften DTU-Modell.
Zwecks Fehlereinkreisung hatte ich noch eine baugleiche DTU eingesetzt. Wieder das gleiche Problem. Daher hatte ich die DTU nicht mehr im Fokus.

Ich habe aus Frust dann auch eine DTU von einem anderen Hersteller gekauft - und siehe da - bis auf mini Ausnahmen läuft sie stabil.
Überhaupt kein Vergleich zur anderen.

Im Router konnte man auch den Vergleich sehr gut erkennen. Was hat mich das Zeit und Nerven gekostet.

Ich bin auf WLAN gar nicht gekommen, weil die DTU 3 Meter von der Fritz Box entfernt war und die Fehler"meldung" eigentlich das Hochzählen der Sekunden war (->DTU zu WR).

Danke Dir nochmals!
